### PR TITLE
[angular] Improve linting

### PR DESCRIPTION
**Pull request by kaidohallik**
Follow up to work done for #12764

I compared our current config with ESLint and TypeScript ESLint recommended configs and removed our identical config and also fixed some recommended rules.

Below is commit history.

Sort ESLint rules alphabetically for better readability.

In the latest ESLint + TypeScript ESLint is `no-unused-vars` identical to our current config, so deleting unnecessary identical override.

Fix `@typescript-eslint/no-unsafe-return` errors.

Fix `@typescript-eslint/restrict-template-expressions` errors.

https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/src/configs/eslint-recommended.ts and https://github.com/eslint/eslint/blob/master/conf/eslint-recommended.js doesn't contain `no-shadow`, so removing `"no-shadow": "off"`.

https://github.com/typescript-eslint/typescript-eslint/blob/master/packages/eslint-plugin/src/configs/eslint-recommended.ts contains already `"no-var": "error"` and `"prefer-const": "error"` - so removing those overrides.

---

Please make sure the below checklist is followed for Pull Requests.

-   [x] [All continuous integration tests](https://github.com/jhipster/generator-jhipster/actions) are green
-   [ ] Tests are added where necessary
-   [ ] [jhipster-online](https://github.com/jhipster/jhipster-online) was updated if necessary
-   [ ] Documentation is added/updated where necessary
-   [ ] Coding Rules & Commit Guidelines as per our [CONTRIBUTING.md document](https://github.com/jhipster/generator-jhipster/blob/main/CONTRIBUTING.md) are followed

When you are still working on the PR, consider converting it to Draft (bellow reviewers) and adding `skip-ci` label, you can still see CI build result at your branch.

### CODE REVIEW

**Review by mshima — APPROVED** [review]

**Comment by kaidohallik** [comment]
I found that for oauth2 user.id type is generated as number, should be string, fixing this shortly.

**Comment by mshima** [comment]
@kaidohallik if you want to ignore User id.
It will be handled soon by composite pk.

**Comment by kaidohallik** [comment]
Fixed `user.id` datatype for `oauth2` now. Just in case used function `getUserPkType` also in user-management tests (although it is not generated in `oauth2`).  
Removed [angular] from PR title to launch full tests as now also back-end part changed as using created `getUserPkType` also in back-end.

File: generators/generator-base-private.js [hunk 1]
```diff
@@ -1262,6 +1262,20 @@ module.exports = class JHipsterBasePrivateGenerator extends Generator {
         return pk;
     }
 
+    /**
+     * Return the User entity primary key data type
+     *
+     * @param {any} databaseType - the database type
+     * @param {any} authenticationType - the authentication type
+     */
+    getUserPkType(databaseType, authenticationType) {
```
Comment (mshima) [inline generators/generator-base-private.js:1271]: The approach needed by composite id is quite different.
This function will be useless soon.
I prefer to don’t add it.

Comment (kaidohallik) [reply generators/generator-base-private.js]: OK, I made minimal changes, changed only `user.model.ts`, rewrote last commit with those minimal changes, restored [angular] in title to run only Angular tests.

**Pull request merged**